### DISCUSSION TITLE
Is this a bug?

### DISCUSSION THREAD

**Discussion by JohnFinn** [google/googletest#4042] [Community Help]
I have a code that looks like this
```c++
class IFoo {
public:
  virtual void foo(int) const =0;
};

class FooMock {
public:
  MOCK_CONST_METHOD1(foo, void(int));
};

TEST(Foo, bar)
{
  using namespace ::testing;
  FooMock foo_mock;
  EXPECT_CALL(foo_mock, foo(Ne(1))).Times(0);
  foo_mock.foo(1);
}
```
When I run tests, it fails with the following is in the output
```
Unexpected mock function call - returning directly.
    Function call: foo(1)
Google Mock tried the following 1 expectation, but it didn't match:
EXPECT_CALL(foo_mock, foo(Ne(1)))...
  Expected arg #0: isn't equal to 1
           Actual: 1
         Expected: to be never called
           Actual: never called - saturated and active
```
The lines `Expected: to be never called` and `Actual: never called` sound like the code is behaving like expected.

The question is - Is this a bug and if not, what is the meaning of the output?

**Comment by Griffon26** [comment]
This is expected behavior. You have made an expectation for calls that have params unequal to 1, but you did not write any expectations about calls that are equal to 1.

GoogleMock is telling you that it did not expect that call foo(1) and that it tried to match it against the expectation you wrote, which happened to be in the state of "not called so far".

If you have any explicit expectations on a function, all calls to that function should have expectations, for example by adding:
EXPECT_CALL(foo_mock, foo(1)).Times(AnyNumber()). In fact if you had only this expectation, then you wouldn't have to specify that any other values are not allowed.
